boringtun/src/device/mod.rs

```diff
@@ -258,32 +268,48 @@ impl DeviceHandle {
     }
 }
 
-impl Drop for DeviceHandle {
+impl<R: Registry> DeviceHandleBuilder<R> {
+    pub fn with_registry(self, registry: R) -> Self {
+        Self {
+            registry: Some(registry),
+            ..self
+        }
+    }
+
+    pub fn build(self) -> Result<DeviceHandle<R>, Error> {
+        let n_threads = self.config.n_threads;
+        let mut wg_interface =
+            Device::new(&self.name, self.config, self.registry.expect("registry"))?;
+        wg_interface.open_listen_socket(0)?; // Start listening on a random port
+
+        let interface_lock = Arc::new(Lock::new(wg_interface));
+
+        let mut threads = vec![];
+
+        for i in 0..n_threads {
+            threads.push({
+                let dev = Arc::clone(&interface_lock);
+                thread::spawn(move || DeviceHandle::event_loop(i, &dev))
+            });
+        }
```

Comment:
Feels like a little refactoring here could allow for async support 🤔

Reply:
Lack of support in general aside, it wouldn't make sense in this case as these are even loops which should be run in different threads.  Switching to async/await would mean that we'd never return from `build`.